qutip/solver/mcsolve.py

```diff
@@ -0,0 +1,311 @@
+            self.target_norm = self.generator.random()
+        self._integrator.set_state(t_guess, state_new)
+
+    def _argument(self, args):
+        if self._integrator:
+            self._integrator.arguments(args)
+        self.rhs.arguments(args)
+        for c_op in self._c_ops:
+            c_op.arguments(args)
+        for n_op in self._n_ops:
+            n_op.arguments(args)
+
+
+# -----------------------------------------------------------------------------
+# MONTE CARLO CLASS
+# -----------------------------------------------------------------------------
+class McSolver(MultiTrajSolver):
+    r"""
+    Monte Carlo Solver of a state vector :math:`|\psi \rangle` for a
+    given Hamiltonian and sets of collapse operators. Options for the
+    underlying ODE solver are given by the Options class.
+
+    Parameters
+    ----------
+    H : :class:`qutip.Qobj`, :class:`qutip.QobjEvo`, ``list``, callable.
+        System Hamiltonian as a Qobj, QobjEvo. It can also be any input type
+        that QobjEvo accepts (see :class:`qutip.QobjEvo`'s documentation).
+        ``H`` can also be a superoperator (liouvillian) if some collapse
+        operators are to be treated deterministically.
+
+    c_ops : list
+        A ``list`` of collapse operators in any input type that QobjEvo accepts
+        (see :class:`qutip.QobjEvo`'s documentation). They must be operators
+        even if ``H`` is a superoperator.
+
+    options : SolverOptions, [optional]
+        Options for the evolution.
+
+    seed : int, SeedSequence, list, [optional]
+        Seed for the random number generator. It can be a single seed used to
+        spawn seeds for each trajectory or a list of seed, one for each
+        trajectory. Seeds are saved in the result and can be reused with::
+            seeds=prev_result.seeds
+    """
+    _traj_solver_class = _McTrajectorySolver
+    name = "mcsolve"
+    optionsclass = SolverOptions
+
+    def __init__(self, H, c_ops, *, options=None, seed=None):
+        _time_start = time()
+
+        if isinstance(c_ops, (Qobj, QobjEvo)):
+            c_ops = [c_ops]
+        c_ops = [QobjEvo(c_op) for c_op in c_ops]
+        if H.issuper:
+            self._c_ops = [
+                spre(c_op) * spost(c_op.dag()) if c_op.isoper else c_op
+                for c_op in c_ops
+            ]
+            self._n_ops = self._c_ops
+            rhs = QobjEvo(H)
+            for c_op in c_ops:
+                cdc = c_op.dag() @ c_op
+                rhs -= 0.5 * (spre(cdc) + spost(cdc))
+        else:
+            self._c_ops = c_ops
+            self._n_ops = [c_op.dag() * c_op for c_op in c_ops]
+            rhs = -1j * QobjEvo(H)
+            for n_op in self._n_ops:
+                rhs -= 0.5 * n_op
+
+        super().__init__(rhs, options=options)
+
+        self.stats['solver'] = "MonteCarlo Evolution"
+        self.stats['num_collapse'] = len(c_ops)
+        self.stats["preparation time"] = time() - _time_start
```

Comment:
preparation time in mesolve is used to measure the start method and not the init. I find this a little bit inconsistent. Did you perhaps change it because it no longer made sense to be defined in the start method?

Reply:
No, it measure both in `mesolve`. `__init__` and `start` and the preparation part of `run` are all included into "preparation time".